src/bits/shared.h

```diff
@@ -53,6 +53,19 @@ namespace detail {
     template <typename FN, typename... ARGS>
     using invoke_result_t = typename invoke_result<FN, ARGS...>::type;
 
+    // Test if two variables is at the same address.
+    template <typename T, typename S>
+    bool is_at_same_address(T, S)
+    {
+        return false;
```

Comment:
This looks wrong, though; two pointers of different apparent type (e.g. base/derived) can point to the same object. I'd just remove this.